src/distilabel/dataset.py

```diff
@@ -90,3 +94,74 @@ def to_argilla(self) -> "FeedbackDataset":
                     stacklevel=2,
                 )
         return rg_dataset
+
+    def save_to_disk(self, dataset_path: PathLike, **kwargs: Any) -> None:
+        """Saves the datataset to disk, also saving the task.
+
+        Args:
+            dataset_path: Path to the dataset.
+            **kwargs: Additional arguments to be passed to `datasets.Dataset.save_to_disk`.
+        """
+        super().save_to_disk(dataset_path, **kwargs)
+        if self.task is not None:
+            save_task_to_disk(dataset_path, self.task)
+
+    @classmethod
+    def load_from_disk(cls, dataset_path: PathLike, **kwargs: Any):
+        """Load a CustomDataset from disk, also reading the task.
+
+        Args:
+            dataset_path: Path to the dataset, as you would do with a standard Dataset.
+
+        Returns:
+            The loaded dataset.
+        """
+        ds = super().load_from_disk(dataset_path, *kwargs)
+        # Dynamically remaps the `datasets.Dataset` to be a `CustomDataset` instance
+        ds.__class__ = cls
+        task = load_task_from_disk(dataset_path)
+        ds.task = task
+        return ds
+
+
+@dataclass
+class DatasetCheckpoint:
+    """A checkpoint class that contains the information of a checkpoint.
+
+    Args:
+        path (Path): The path to the checkpoint.
+        save_frequency (int): The frequency at which the checkpoint should be saved
+            By default is set to -1 (no checkpoint is saved to disk, but the dataset
+            is returned upon failure).
+        extra_kwargs (dict[str, Any]): Additional kwargs to be passed to the `save_to_disk` method of the Dataset.
+
+    Examples:
+        >>> from distilabel.dataset import DatasetCheckpoint
+        >>> # Save the dataset every 10% of the records generated.
+        >>> checkpoint = DatasetCheckpoint(save_frequency=len(dataset) // 10)
+        >>> # Afterwards, we can access the checkpoint checkpoint.path.
+    """
+
+    path: Path = Path.cwd() / "dataset_checkpoint"
```

Comment:
Maybe we can follow the standard and name this `ckpt/` instead?

Reply:
Sure!